src/pg/connection.cr

```diff
@@ -37,6 +37,16 @@ module PG
       end
     end
 
+    # `#initialize` Connect to the server with values of Hash.
+    #
+    # ```
+    # info = { host: "localhost", user: "postgres", password: "password", db_name: "test_db", port: "5432"}
+    # PG::Connection.new(info)
+    # ```
+    def initialize(parameters : Hash(YAML::Type, YAML::Type))
```

Comment:
if the parameters type is `Hash(String, YAML::Type)`,  error will happen.
error like `can't cast (Nil | String | Array(YAML::Type) | Hash(YAML::Type, YAML::Type)) to Hash(String, YAML::Type)`

    y = YAML.load("host: localhost\ndbname: postgres\nuser: postgres\npassword: postgres") as Hash
    PG.connect(y)

It is why I set the parameters type is `Hash(YAML::Type, YAML::Type)`.